homeassistant/components/light/tcpbulbs.py

```diff
@@ -0,0 +1,156 @@
+"""
+
+Support for TCP Connected (Greenwave Reality) lights.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tcpbulbs/
+
+"""
+import logging
+import voluptuous as vol
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, Light, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS)
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+SUPPORTED_FEATURES = (SUPPORT_BRIGHTNESS)
+_LOGGER = logging.getLogger(__name__)
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+
+def pullXML(host, bulbxml):
+    """Pull XML Data from Gateway."""
+    global REQUIREMENTS
+    REQUIREMENTS = ['xmltodict']

```

Comment:
Requirements should really be defined before any function and not inside of it.

Reply:
[Here](https://home-assistant.io/developers/code_review_platform/#1-requirements) it says "Requirements should only be imported inside functions. This is necessary because requirements are installed on the fly."